src/json2typescript/json-convert.ts

```diff
@@ -433,7 +433,7 @@ export class JsonConvert {
      *
      * @see https://www.npmjs.com/package/json2typescript full documentation
      */
-    deserialize<T>(json: any, classReference: { new(): T }): T | T[] {
+    deserialize<T extends object>(json: any, classReference: { new(): T }): T | T[] {
```

Comment:
Is it good to say `T extends object` or can we go even further?

Reply:
Seems like T extends object is probably good enough?  Presumably it should only be restricted to non-primitives, but otherwise put no constraints on the type of object.

Reply:
> Seems like T extends object is probably good enough?  Presumably it should only be restricted to non-primitives, but otherwise put no constraints on the type of object.

I think it depends on the parameter you are passing. If I recall correcty, the serializaion sometimes keeps the original instance and just reassigns values. I will check later if that is true.